### PR TITLE
[BUILD-473] Project Scaffolding

### PR DESCRIPTION
**Pull request by otaviof**
Adding initial components to this project:
- Hack scripts and `Makefile`
- GitHub Action automation
- Helm Chart

### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by otaviof** [comment]
@adambkaplan, @coreydaley could I have permissions to label PRs on this repository, please? (thanks in advance)

**Comment by adambkaplan** [comment]
@otaviof the `kind/*` labels get applied with the `/kind` prow command.

**Review by adambkaplan — COMMENTED** [review]
This kind of conflicts with #1 - not completely, but there are enough overlaps that will result in a merge conflict. I like this has the helm chart deployment option, and the scripts to deploy Tekton and a test registry.

We at least need to decide which one merges first.

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  test-unit:
+    name: test-unit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup
+
+      - name: test-unit
+        run: |
+          make test-unit
```
Comment (coreydaley) [inline .github/workflows/test.yaml:24]:
```suggestion
        run: make test-unit
```

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    name: test-e2e
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup
+      - uses: ./.github/actions/shipwright
+
+      - name: test-e2e
+        run: |
+          make test-e2e
```
Comment (coreydaley) [inline .github/workflows/test.yaml:36]:
```suggestion
        run: make test-e2e
```

File: .gitignore [hunk 3]
```diff
@@ -1,15 +1,2 @@
...
-
-# Test binary, built with `go test -c`
 *.test
-
-# Output of the go coverage tool, specifically when used with LiteIDE
-*.out
-
-# Dependency directories (remove the comment below to include it)
-# vendor/
+triggers
```
Comment (coreydaley) [inline .gitignore:2]: What exactly is this ignoring?

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		./chart | \
+			ko apply ${KO_OPTS} --filename -
+
+.PHONY: test-unit
+test-unit:
+	go test $(GOFLAGS_TEST) ./...
+
+.PHONY: test-e2e
+test-e2e:
+	echo "TODO: Run 'make test-e2e' target!"
```
Comment (coreydaley) [inline Makefile:42]:
```suggestion
	echo "Not implemented"
```

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
```
Comment (coreydaley) [inline Makefile:43]:
```suggestion
.PHONY: verify-kind
```

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+verify-kind:
+	./hack/verify-kind.sh
+
```
Comment (coreydaley) [inline Makefile:46]:
```suggestion

.PHONY: install-registry
```

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+install-registry:
+	./hack/install-registry.sh
+
```
Comment (coreydaley) [inline Makefile:49]:
```suggestion

.PHONY: install-shipwright
```

File: Makefile [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+NAMESPACE ?= shipwright-build
+
+KO_DOCKER_REPO ?= $(IMAGE_BASE)
+KO_OPTS ?= --base-import-paths --tags=${IMAGE_TAG}
+
+.EXPORT_ALL_VARIABLES:
+
+.PHONY: $(APP)
+$(APP):
+	go build .
```
Comment (coreydaley) [inline Makefile:18]:
```suggestion
	go build . -race
```
Might be worth adding the `-race` flag

**Review by coreydaley — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 3 above)
Comment (otaviof) [reply .gitignore:2]: That ignores the executable produced by `go build .` command.

**Comment by coreydaley** [comment]
You should check the shipwright-io/build repository and see if there are other files that you should have in this repository such as https://github.com/shipwright-io/build/blob/main/code-of-conduct.md and a CONTRIBUTING.md

File: main.go [hunk 9]
```diff
@@ -0,0 +1,5 @@
+package main
+
+func main() {
+
```
Comment (coreydaley) [inline main.go:4]: Maybe add something like printing `Welcome to Shipwright Triggers` to stdout.

**Comment by coreydaley** [comment]
A couple of small things but otherwise looks good to me.

File: main.go (hunk 9 above)
Comment (otaviof) [reply main.go:4]: Sure!

**Comment by coreydaley** [comment]
/lgtm
Looks like it still needs a release note, but otherwise it looks good to me. We can always iterate in future pull requests if needed.

**Pull request merged**